src/basic/time-util.c

```diff
@@ -362,6 +363,13 @@ char *format_timestamp_style(
         if (!localtime_or_gmtime_r(&sec, &tm, utc))
                 return NULL;
 
+        if (style == TIMESTAMP_UNIX) {
+                if (strftime(buf, l, "@%s", &tm) <= 0)
+                        return NULL; /* Doesn't fit */
```

Comment:
hmm, so here you convert microseconds since 1970 to "broken down" (i.e. calendar) time first, via `localtime_or_gmtime()` and then convert it back to seconds since 1970 via `strftime()`'s `%s` specifier. Please just use the value as is, i.e. add right after the timestamp-is-unset check above just:

```c
r = snprintf(buf, l, "@" FMT_USEC, t / USEC_PER_SEC);  /* round down µs → s */
if (r < 0 || (size_t) r >= l)        
        return NULL;
```

That way we don't have do the roundtripping through the calendar time. Which should make things a bit more robust, if between the forth and back conversion the timezone or summer time changes or so.